test/powershell/Modules/Microsoft.WSMan.Management/ConfigProvider.Tests.ps1

```diff
@@ -0,0 +1,478 @@
+                Remove-Item WSMan:\localhost\Plugin\TestPlugin2\
+                "WSMan:\localhost\Plugin\TestPlugin2" | Should Not Exist
+            }
+        }
+
+        It "New-Item and Remove-Item for a plugin using a file" {
+            $fileXml = @"
+<PlugInConfiguration Name="TestPlugin2" Filename="%windir%\system32\pwrshplugin.dll" SDKVersion="2" XmlRenderingType="text" Enabled="false"
+ Architecture="64" UseSharedProcess="true" ProcessIdleTimeoutSec="0" RunAsUser="" RunAsPassword="" AutoRestart="false" RunAsVirtualAccount="false"
+ RunAsVirtualAccountGroups="" OutputBufferingMode="Block" xmlns="http://schemas.microsoft.com/wbem/wsman/1/config/PluginConfiguration">
+    <InitializationParameters>
+        <Param Name="PSVersion" Value="5.1"></Param>
+    </InitializationParameters>
+    <Resources>
+        <Resource ResourceUri="http://schemas.microsoft.com/powershell/TestPlugin2" SupportsOptions="true" ExactMatch="true">
+            <Security Uri="http://schemas.microsoft.com/powershell/TestPlugin2"
+                Sddl="O:NSG:BAD:P(A;;GA;;;BA)(A;;GA;;;IU)(A;;GA;;;RM)S:P(AU;FA;GA;;;WD)(AU;SA;GXGW;;;WD)" ExactMatch="False"></Security>
+            <Capability Type="Shell"></Capability>
+        </Resource>
+    </Resources>
+    <Quotas MaxIdleTimeoutms="2147483646" MaxConcurrentUsers="2147483646" IdleTimeoutms="6200000" MaxProcessesPerShell="2147483646"
+        MaxMemoryPerShellMB="2147483646" MaxConcurrentCommandsPerShell="2147483646" MaxShells="2147483646" MaxShellsPerUser="2147483646"></Quotas>
+</PlugInConfiguration>
+"@
+            Set-Content -Path $testdrive\plugin.xml -Value $fileXml
+            try {
+                $plugin = New-Item -Path WSMan:\localhost\Plugin -File $testdrive\plugin.xml -Name TestPlugin2
+                Test-Plugin -Plugin $plugin
+            }
+            finally {
+                Remove-Item WSMan:\localhost\Plugin\TestPlugin2\
+                "WSMan:\localhost\Plugin\TestPlugin2\" | Should Not Exist
+            }
+        }
+
+        It "New-Item and Remove-Item for a plugin resource" {
+            try {
+                $resource = New-Item -Path WSMan:\localhost\Plugin\TestPlugin\Resources\ `
+                    -ResourceUri http://foo -Capability shell
+                $resource.PSPath | Should Exist
+                $properties = Get-ChildItem $resource.PSPath
+                ($properties | Where-Object { $_.Name -eq "ResourceUri" }).Value | Should Be "http://foo/"
+                ($properties | Where-Object { $_.Name -eq "Capability" })[0].Value | Should Be "shell"
+            }
+            finally {
+                Remove-Item $resource.PSPath
+                $resource.PSPath | Should Not Exist
+            }
+        }
+
+        It "New-Item and Remove-Item for an initialization parameter" {
+            try {
+                $parameter = New-Item -Path WSMan:\localhost\Plugin\TestPlugin\InitializationParameters `
+                    -ParamName foo -ParamValue bar
```

Comment:
We should not use "foo" and "bar" since they violate PolyCheck. Please use other string values.

Reply:
I believe policheck allows it for code, but not in strings given to users.  In any case, if we need to fix this, we should have a separate PR for all the code.